src/factory-reset/factory-reset-tool.c

```diff
@@ -0,0 +1,380 @@
+#include "os-util.h"
+#include "pretty-print.h"
+#include "udev-util.h"
+#include "varlink-io.systemd.FactoryReset.h"
+#include "varlink-util.h"
+#include "verbs.h"
+
+static bool arg_retrigger = false;
+static bool arg_quiet = false;
+static bool arg_varlink = false;
+
+static int help(void) {
+        _cleanup_free_ char *link = NULL;
+        int r;
+
+        r = terminal_urlify_man("systemd-factory-reset", "8", &link);
+        if (r < 0)
+                return log_oom();
+
+        printf("%1$s [OPTIONS...] COMMAND\n"
+               "\n%5$sQuery, request, cancel factory reset operation.%6$s\n"
+               "\n%3$sCommands:%4$s\n"
+               "  status             Report current factory reset status\n"
+               "  request            Request a factory reset on next boot\n"
+               "  cancel             Cancel a prior factory reset request for next boot\n"
+               "  complete           Mark a factory reset as complete\n"
+               "\n%3$sOptions:%4$s\n"
+               "  -h --help          Show this help\n"
+               "     --version       Print version\n"
+               "     --retrigger     Retrigger block devices\n"
+               "  -q --quiet         Suppress output\n"
+               "\nSee the %2$s for details.\n",
+               program_invocation_short_name,
+               link,
+               ansi_underline(),
+               ansi_normal(),
+               ansi_highlight(),
+               ansi_normal());
+
+        return 0;
+}
+
+static int parse_argv(int argc, char *argv[]) {
+        enum {
+                ARG_PATH = 0x100,
+                ARG_VERSION,
+                ARG_RETRIGGER,
+        };
+
+        static const struct option options[] = {
+                { "help",      no_argument, NULL, 'h'           },
+                { "version",   no_argument, NULL, ARG_VERSION   },
+                { "retrigger", no_argument, NULL, ARG_RETRIGGER },
+                { "quiet",     no_argument, NULL, 'q'           },
+                {}
+        };
+
+        int r, c;
+
+        assert(argc >= 0);
+        assert(argv);
+
+        while ((c = getopt_long(argc, argv, "hq", options, NULL)) >= 0)
+                switch (c) {
+
+                case 'h':
+                        return help();
+
+                case ARG_VERSION:
+                        return version();
+
+                case ARG_RETRIGGER:
+                        arg_retrigger = true;
+                        break;
+
+                case 'q':
+                        arg_quiet = true;
+                        break;
+
+                case '?':
+                        return -EINVAL;
+
+                default:
+                        assert_not_reached();
+                }
+
+        r = sd_varlink_invocation(SD_VARLINK_ALLOW_ACCEPT);
+        if (r < 0)
+                return log_error_errno(r, "Failed to check if invoked in Varlink mode: %m");
+        if (r > 0)
+                arg_varlink = true;
```

Comment:
`arg_varlink = r > 0`?

Reply:
i want to keep the door open to enter this mode by other means, i.e. env var or cmdline switch, and we are doing it the same in various other places.